src/page/MainScreen/MainScreen.test.tsx

```diff
@@ -56,6 +64,27 @@ describe('MainScreen', () => {
     getByTestId(`Main-AnimatedCard-${mockListMovies[0].imdbID}`);
   });
 
+  test('should renders new movies by changing page', async () => {
+    mockedMovieDeck.mockImplementation(({handlePage}: MovieDeckProps) => {
+      const {Button} = require('react-native');
+      return <Button title="button" testID="changePage" onPress={handlePage} />;
+    });
+
+    const mockUseQuery = {
+      data: [],
+      isLoading: true,
+      error: null,
+    };
+    mockedUseQuery.mockReturnValue(
+      mockUseQuery as UseQueryResult<unknown, unknown>,
+    );
```

Comment:
Did you read about [testing network calls with react-query](https://tanstack.com/query/latest/docs/framework/react/guides/testing)? Not required, yet if seted up properly it would be easier to test queries.